main/helpers/messages.ts

```diff
@@ -17,65 +22,104 @@ export interface AnswerMessage extends ClientMessage {
 	answer: string
 }
 
-export interface ServerMessage {
-	type: "pong" | "throwDice" | "registered"
+export interface ACKMessage extends ClientMessage {
+	type: "ACK"
+	text: string
 }
 
-export type TurnMessage = {
-	type: "yourTurn"
-	nick: string
+export interface QuestionMessage extends ClientMessage {
+	type: "question"
+	content: string
+	answers: string[]
 }
 
-// This should not stay here
-export let clients = new Map<string, WebSocket>()
 
 export const handleMessage = (msg: ClientMessage, ws: WebSocket) => {
+	
 	switch (msg.type) {
 		case "ping":
-			handlePing(ws)
-			// throwDice("test")
-			break
+			const from = websocketsToNick.get(ws);
+			const to = clients.get(msg.destination)
+			to?.send(JSON.stringify({ type: "ping", from: from }))
+			break;
 
 		case "register":
-			const registerMsg = msg as RegisterMessage
-			handleRegister(registerMsg, ws)
-			break
+			const registerMsg = msg as RegisterMessage;
+			handleRegister(registerMsg, ws);
+			break;
 
 		case "dice":
-			const diceMsg = msg as DiceThrowMessage
-			handleDiceThrow(diceMsg)
-			break
+			const diceMsg = msg as DiceThrowMessage;
+			handleDiceThrow(diceMsg, ws);
+			break;
 
 		case "answer":
-			const answerMsg = msg as AnswerMessage
-			handleAnswer(answerMsg)
-			break
-
+			const answerMsg = msg as AnswerMessage;
+			handleAnswer(answerMsg, ws);
+			break;
+
+		case "pong":
+			const pongMsg  = msg as ClientMessage
+			console.log("Pong from App, message : received ping from " + pongMsg.destination)
+			break;
+
+		case "ACK":
+			const ACKMessage = msg as ACKMessage
+			console.log("Received ACK from app, message : " + ACKMessage.text)
+			break;
+
+		case "question":
+			const questionMessage = msg as QuestionMessage
+			sendQuestion(questionMessage)
+			break;
+
+		case "throwDice":
+			throwDice(msg.destination)
+			break;
+			
 		default:
-			console.error("Unknown message type", JSON.stringify(msg))
+			console.error("Unknown message type", JSON.stringify(msg));
 	}
 }
 
 export const throwDice = (nick: string) => {
 	const ws = clients.get(nick)
-	ws.send(JSON.stringify({ type: "throwDice" }))
+	ws?.send(JSON.stringify({ type: "throwDice" }))
```

Comment:
```suggestion
	ws!.send(JSON.stringify({ type: "throwDice" }))
```
Myśle że w tym miejscu lepiej żeby coś się wywaliło niż po prostu się nie wykonało